nailgun/entities.py

```diff
@@ -3953,6 +4178,39 @@ def refresh(self, synchronous=True, **kwargs):
         response = client.put(self.path('refresh'), **kwargs)
         return _handle_response(response, self._server_config, synchronous)
 
+    def import_puppetclasses(self, synchronous=True, **kwargs):
+        """Import puppet classes from puppet Capsule.
+
+        :param synchronous: What should happen if the server returns an HTTP
+            202 (accepted) status code? Wait for the task to complete if
+            ``True``. Immediately return the server's response otherwise.
+        :param kwargs: Arguments to pass to requests.
+        :returns: The server's response, with all JSON decoded.
+        :raises: ``requests.exceptions.HTTPError`` If the server responds with
+            an HTTP 4XX or 5XX message.
+
+        """
+        kwargs = kwargs.copy()
+        kwargs.update(self._server_config.get_client_kwargs())
+        # Check if environment_id was sent and substitute it to the path
+        # but do not pass it to requests
+        if 'environment' in kwargs:
+            if isinstance(kwargs['environment'], Environment):
+                environment_id = kwargs.pop('environment').id
+            else:
+                environment_id = kwargs.pop('environment')
+            response = client.post(
+                '{0}/environments/{1}/import_puppetclasses'.format(
+                    self.path(), environment_id),
+                **kwargs
+            )
+        else:
+            response = client.post(
+                '{0}/import_puppetclasses'.format(self.path()),
+                **kwargs
+            )
```

Comment:
Nitpick: i'd rather form 2 different pathes in this `if..else` condition and used `client.post` only once with the final path. It's not a blocker, just prefer the least possible code duplication.